dali/kernels/signal/window/extract_windows_gpu.h

```diff
@@ -0,0 +1,104 @@
+// Copyright (c) 2019, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_KERNELS_SIGNAL_WINDOW_EXTRACT_WINDOWS_GPU_H_
+#define DALI_KERNELS_SIGNAL_WINDOW_EXTRACT_WINDOWS_GPU_H_
+
+#include <memory>
+#include "dali/core/convert.h"
+#include "dali/kernels/signal/window/extract_windows_args.h"
+#include "dali/kernels/kernel.h"
+
+namespace dali {
+namespace kernels {
+namespace signal {
+
+struct ExtractWindowsBatchedArgs : ExtractWindowsArgs {
+  bool vertical = false;
+  /**
+   * @brief If true, all outputs are concatenated.
+   *
+   * In case of vertical windows, tThe concatenated output will contain all first samples from
+   * windows from 1st recording, then 2nd, etc, and then all seconds samples from all
+   * recordings and so forth.
+   * For horizontal windows, the concatenation is trivial.
+   */
+  bool concatenate = true;
+  /**
+   * @brief Indicates that the output should be overallocated (or windows truncated) to this size.
+   */
+  int padded_output_window = -1;
```

Comment:
Does it map to `out_win_length`?

Reply:
Yes. Neither name really reflects what it does... it's hard make a descriptive name for it, really.

Reply:
If we can't figure the best name, at least use one everywhere.